src/components/Board/Board.js

```diff
@@ -50,63 +46,71 @@ const Board = () => {
       player2Pos,
       p1StartPermission,
       p2StartPermission,
-      isSnake,
       snake,
-      isLadder,
       ladder,
       buttonAbility,
     },
     dispatch,
   ] = React.useReducer(reducer, initialStates);
 
   // I believe this should be in a useEffect since it is actually a side effect :)
+  //Done:)
+  //******************************************* QUESTION ********************************************
+  //Why is this considered a side effect? because the functionality depends on these two dependencies? or there's another reason?
+
   // Checking if any of the players reaches 100, to announce the game finished.
```

Comment:
These lines:
```js
if (!gameOver && (player1Pos === 100 || player2Pos === 100)) {
    dispatch({ type: "gameOver" });
  }
```

These will be executed every time there is a re-render in this component.
We only want to execute this based on some criterias.
You have perfectly defined it, but if you don't wrap it into a useEffect, as I mentioned above your application will try to execute these lines. In our case, this is not an expensive operation (fortunately) so it won't cause any significant performance problems, but it is not the best approach IMO.
 
 Regarding your question about why it is a side effect:
(Even myself struggle with this term sometimes haha)
 I consider this as a side effect, because these lines are modifying the entire application's state and ends the game.